File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        return self.binary(0, len(nums)-1, nums, target)
    def binary(self, l, r, nums, target):
        if r < l:
            return -1
        mid = l + (r-l) // 2
        if nums[mid] == target:
            return mid
        elif nums[mid] < target:
            return self.binary(mid+1, r, nums, target)
        else:
            return self.binary(l, mid-1, nums, target)

File: 0704-binary-search/0704-binary-search.go
func search(nums []int, target int) int {
    l := 0
    r := len(nums) - 1
    
    for l <= r {
        mid := l + (r - l) // 2
        if nums[mid] == target {
            return mid
        } else if nums[mid] < target {
            l = mid + 1
        } else {
            r = mid - 1
        }
    }
    return -1
}